internal/provider/kubernetes/controller.go

```diff
@@ -1105,6 +1105,7 @@ func (r *gatewayAPIReconciler) watchResources(ctx context.Context, mgr manager.M
 	// Watch HTTPRoute CRUDs and process affected Gateways.
 	httprPredicates := []predicate.TypedPredicate[*gwapiv1.HTTPRoute]{
 		predicate.TypedGenerationChangedPredicate[*gwapiv1.HTTPRoute]{},
+		predicate.TypedLabelChangedPredicate[*gwapiv1.HTTPRoute]{},
```

Comment:
can we add this for all xRoutes and Gateway as well ?

Reply:
added @arkodg